jolly-jellyfish/src/page_maker/views.py

```diff
@@ -1,3 +1,242 @@
-from django.shortcuts import render
+from django.urls import reverse_lazy
+from django.shortcuts import get_object_or_404
+from django.core.exceptions import PermissionDenied
+from django.contrib.auth.views import redirect_to_login
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
+from django.views.generic import (
+    TemplateView,
+    FormView,
+    DetailView,
+    UpdateView,
+    DeleteView,
+    CreateView,
+    ListView
+)
 
-# Create your views here.
+from .models import User, Webpage, Template, Comment
+from .forms import UserRegisterForm, WebpageForm, TemplateForm, CommentForm
+
+
+class MainView(TemplateView):
+    template_name = "page_maker/main.html"
+
+
+class UserRegister(FormView):
+    template_name = 'registration/register.html'
+    form_class = UserRegisterForm
+    success_url = '/users/login'
+
+    def form_valid(self, form):
+        form.save()
+        return super().form_valid(form)
+
+
+class UserDetailView(DetailView):
+    model = User
+    context_object_name = 'viewed_user'
+    template_name = 'page_maker/user_detail.html'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        # user = get_object_or_404(User, pk=self.kwargs.get('pk'))
+        user = self.get_object()
+        context['user_pages'] = Webpage.objects.filter(author=user)
+        return context
+
+
+class UserUpdateView(LoginRequiredMixin, UserPassesTestMixin, UpdateView):
+    model = User
+    template_name = 'page_maker/user_update.html'
+    fields = ['first_name', 'last_name', 'email']
+    success_url = '/'
+
+    def test_func(self):
+        user = self.get_object()
+        if self.request.user == user:
+            return True
+        return False
+
+
+class UserDeleteView(LoginRequiredMixin, UserPassesTestMixin, DeleteView):
+    model = User
+    template_name = 'page_maker/user_delete.html'
+    success_url = '/'
+
+    def test_func(self):
+        user = self.get_object()
+        if self.request.user == user:
+            return True
+        return False
+
+
+class WebpageCreateView(LoginRequiredMixin, FormView):
+    template_name = 'page_maker/webpage_create.html'
+    form_class = WebpageForm
+    # success_url = '/pages'
+
+    # TODO validate and sanitize text and images
```

Comment:
How much of this does Django do for us?

Reply:
It probably handles sanitizing both text and images, I removed the comment.